lib/src/editorhelper.cpp

```diff
@@ -247,42 +377,44 @@ void apply_crop(uhdr_raw_image_t* src, int left, int top, int wd, int ht) {
   src->h = ht;
 }
 
-std::unique_ptr<uhdr_raw_image_ext_t> apply_resize(uhdr_raw_image_t* src, int dst_w, int dst_h) {
+std::unique_ptr<uhdr_raw_image_ext_t> apply_resize(ultrahdr::uhdr_resize_effect_t* desc,
+                                                   uhdr_raw_image_t* src, int dst_w, int dst_h) {
   std::unique_ptr<uhdr_raw_image_ext_t> dst = std::make_unique<uhdr_raw_image_ext_t>(
       src->fmt, src->cg, src->ct, src->range, dst_w, dst_h, 1);
 
   if (src->fmt == UHDR_IMG_FMT_24bppYCbCrP010) {
     uint16_t* src_buffer = static_cast<uint16_t*>(src->planes[UHDR_PLANE_Y]);
     uint16_t* dst_buffer = static_cast<uint16_t*>(dst->planes[UHDR_PLANE_Y]);
-    resize_buffer(src_buffer, dst_buffer, src->w, src->h, dst->w, dst->h, src->stride[UHDR_PLANE_Y],
-                  dst->stride[UHDR_PLANE_Y]);
+    desc->m_resize_uint16_t(src_buffer, dst_buffer, src->w, src->h, dst->w, dst->h,
+                            src->stride[UHDR_PLANE_Y], dst->stride[UHDR_PLANE_Y]);
     uint32_t* src_uv_buffer = static_cast<uint32_t*>(src->planes[UHDR_PLANE_UV]);
     uint32_t* dst_uv_buffer = static_cast<uint32_t*>(dst->planes[UHDR_PLANE_UV]);
-    resize_buffer(src_uv_buffer, dst_uv_buffer, src->w / 4, src->h / 2, dst->w / 4, dst->h / 2,
-                  src->stride[UHDR_PLANE_UV] / 2, dst->stride[UHDR_PLANE_UV] / 2);
+    desc->m_resize_uint32_t(src_uv_buffer, dst_uv_buffer, src->w / 4, src->h / 2, dst->w / 4,
```

Comment:
Should source and dest width be divided by 2 instead of 4? Could you double check?

Reply:
You are right dichen, this looks like mistake. I have fixed it.